### PR TITLE
Multiple subapps can mount

### LINKED ISSUE #14: multiple sub-apps cannot mount on the same root
```js
import { App } from 'https://denopkg.com/deno-libs/tinyhttp@master/mod.ts'
import { makeFetch } from 'https://deno.land/x/superfetch@1.0.4/mod.ts'

const app = new App()

const route1 = new App()
route1.get('/route1', (_req, res) => void res.end('route1'))

const route2 = new App()
route2.get('/route2', (_req, res) => void res.end('route2'))

app.use(route1)
app.use(route2)

const res1 = await makeFetch(app.handler)('/route1')
res1.expect('route1')

const res2 = await makeFetch(app.handler)('/route2')
res2.expect('route2')
```

### PR DESCRIPTION
**Pull request by aarontravass**
fixes #14

### CODE REVIEW

File: dev_deps.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export { makeFetch } from 'https://deno.land/x/superfetch@1.0.4/mod.ts'
+export { makeFetch } from 'https://raw.githubusercontent.com/deno-libs/superfetch/a42dd5ce9fdc1b598e1f22ac4caffb10ec2df4ed/mod.ts'
```
Comment (talentlessguy) [inline dev_deps.ts:1]: please don't import from GitHub

**Comment by aarontravass** [comment]
@talentlessguy should we move to regexparam? https://deno.land/x/regexparam@v2.0.1

File: deps.ts [hunk 2]
```diff
@@ -32,3 +32,4 @@ export {
   STATUS_TEXT,
 } from 'https://deno.land/std@0.185.0/http/http_status.ts'
 export { CookieMap } from 'https://deno.land/std@0.185.0/http/cookie_map.ts'
+export { parse as rg } from 'https://deno.land/x/regexparam@v2.0.1/src/index.js'
```
Comment (talentlessguy) [inline deps.ts:35]: deno version uses URLPattern instead of rg

File: deno.jsonc [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "semiColons": false
   },
   "tasks": {
-    "test": "deno test --allow-net --allow-read --coverage=coverage",
```
Comment (talentlessguy) [inline deno.jsonc:10]: deno task test is used in GH actions, please keep it

**Review by talentlessguy — CHANGES_REQUESTED** [review]

File: dev_deps.ts (hunk 1 above)
Comment (aarontravass) [reply dev_deps.ts:1]: did you publish the latest change?

File: deno.jsonc (hunk 3 above)
Comment (aarontravass) [reply deno.jsonc]: my bad.

File: tests/modules/req.test.ts [hunk 4]
```diff
@@ -292,7 +292,7 @@ describe('Request extensions', () => {
     })
   })
   describe('Network extensions', () => {
```
Comment (aarontravass) [inline tests/modules/req.test.ts:294]: This test was hanging so I commented it out

File: app.ts [hunk 5]
```diff
@@ -311,21 +307,24 @@ export class App<
   /**
    * Creates HTTP server and dispatches middleware
    * @param port server listening port
-   * @param Server callback after server starts listening
    * @param host server listening host
+   * @param Server callback after server starts listening
    */
-  async listen(port: number, cb?: () => void, hostname?: string) {
+  async listen(port: number, hostname?: string, cb?: (error: any) => void) {
```
Comment (talentlessguy) [inline app.ts:313]: why did you change function signature? It should be the same as in node version:

```ts
listen(port?: number, cb?: () => void, host?: string): Server
```

**Review by talentlessguy — CHANGES_REQUESTED** [review]

File: app.ts (hunk 5 above)
Comment (aarontravass) [reply app.ts]: Normally, cbs tend to be the last argument. Such as `emitter.on` in node

Comment (aarontravass) [reply app.ts]: fixed it nonetheless.
